src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1714,6 +1726,139 @@ IOperation negateNullable(IOperation operand)
             }
         }
 
+        private IOperation VisitObjectBinaryConditionalOperator(IBinaryOperation binOp, int? captureIdForResult)
+        {
+            SpillEvalStack();
+
+            Compilation compilation = ((Operation)binOp).SemanticModel.Compilation;
+            INamedTypeSymbol booleanType = compilation.GetSpecialType(SpecialType.System_Boolean);
+            IOperation left = binOp.LeftOperand;
+            IOperation right = binOp.RightOperand;
+            IOperation condition;
+
+            bool isAndAlso = CalculateAndOrSense(binOp, true);
+
+
+            var done = new BasicBlock(BasicBlockKind.Block);
+            var checkRight = new BasicBlock(BasicBlockKind.Block);
+
+            condition = new ConversionOperation(Visit(left), ConvertibleConversion.Instance, isTryCast: false, isChecked: false,
+                                                semanticModel: null, left.Syntax, booleanType, constantValue: default, isImplicit: true);
+
+            LinkBlocks(CurrentBasicBlock, (Operation.SetParentOperation(condition, null), JumpIfTrue: isAndAlso, RegularBranch(checkRight)));
+            _currentBasicBlock = null;
+
+            int resultId = _availableCaptureId++;
+            AddStatement(new FlowCapture(resultId, binOp.Syntax, new LiteralExpression(semanticModel: null, left.Syntax, booleanType, constantValue: !isAndAlso, isImplicit: true)));
+            LinkBlocks(CurrentBasicBlock, done);
+            _currentBasicBlock = null;
+
+            AppendNewBlock(checkRight);
+
+            condition = new ConversionOperation(Visit(right), ConvertibleConversion.Instance, isTryCast: false, isChecked: false,
+                                                semanticModel: null, right.Syntax, booleanType, constantValue: default, isImplicit: true);
+
+            AddStatement(new FlowCapture(resultId, binOp.Syntax, condition));
+
+            AppendNewBlock(done);
+
+            return new ConversionOperation(new FlowCaptureReference(resultId, binOp.Syntax, booleanType, constantValue: default),
+                                           ConvertibleConversion.Instance, isTryCast: false, isChecked: false,
+                                           semanticModel: null, binOp.Syntax, binOp.Type, binOp.ConstantValue, isImplicit: true);
```

Comment:
>binOp.ConstantValue [](start = 90, length = 19)

Is it possible to have a constant value here? 
If so, wouldn't that mean that either the left or the right had a constant value, which should be reflected in the conversions above (line 1746 and 1759)?

Reply:
> Is it possible to have a constant value here? 

No, I do not think an expression of type Object can have a boolean constant value.

---
In reply to: [191549014](https://github.com/dotnet/roslyn/pull/27155#discussion_r191549014) [](ancestors = 191549014)

Reply:
Consider using `constantValue: default` then. Thanks

Reply:
> Consider using constantValue: default then. 

We prefer using information from original nodes rather than making guesses in some particular cases, but not in others.

---
In reply to: [191581103](https://github.com/dotnet/roslyn/pull/27155#discussion_r191581103) [](ancestors = 191581103)